bundles/org.openhab.binding.vdr/src/main/java/org/openhab/binding/vdr/internal/VDRHandler.java

```diff
@@ -0,0 +1,357 @@
+                            }
+                            break;
+                        case "diskUsage":
+                            SVDRPDiskStatus diskStatus = con.getDiskStatus();
+                            result = new DecimalType(diskStatus.getPercentUsed());
+                            break;
+                        case "currentEventTitle":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NOW);
+                            result = new StringType(entry.getTitle());
+                            break;
+                        case "currentEventSubTitle":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NOW);
+                            result = new StringType(entry.getSubtitle());
+                            break;
+                        case "currentEventDuration":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NOW);
+                            result = new DecimalType(entry.getDuration());
+                            break;
+                        case "currentEventBegin":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NOW);
+                            result = new DateTimeType(LocalDateTime
+                                    .ofInstant(entry.getBegin().toInstant(), TimeZone.getDefault().toZoneId())
+                                    .atZone(TimeZone.getDefault().toZoneId()));
+                            break;
+                        case "currentEventEnd":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NOW);
+                            result = new DateTimeType(LocalDateTime
+                                    .ofInstant(entry.getEnd().toInstant(), TimeZone.getDefault().toZoneId())
+                                    .atZone(TimeZone.getDefault().toZoneId()));
+                            break;
+                        case "nextEventTitle":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NEXT);
+                            result = new StringType(entry.getTitle());
+                            break;
+                        case "nextEventSubTitle":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NEXT);
+                            result = new StringType(entry.getSubtitle());
+                            break;
+                        case "nextEventDuration":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NEXT);
+                            result = new DecimalType(entry.getDuration());
+                            break;
+                        case "nextEventBegin":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NEXT);
+                            result = new DateTimeType(LocalDateTime
+                                    .ofInstant(entry.getBegin().toInstant(), TimeZone.getDefault().toZoneId())
+                                    .atZone(TimeZone.getDefault().toZoneId()));
```

Comment:
The user can set a different time zone via the UI. You can retrieve it by injecting the `TimeZoneProvider`.

Reply:
Thanks. Much better!